projects/step-frontend/src/lib/modules/_common/interceptors/http-error.interceptor.ts

```diff
@@ -11,18 +18,41 @@ export class HttpErrorInterceptor implements HttpInterceptor {
   private handleHttpError(error: HttpErrorResponse): Observable<any> {
     console.error('network error', error);
     if (error.status !== 401) {
-      if (error.error?.errorName && error.error?.errorMessage) {
-        this._snackBar.open(error.error.errorName + ': ' + error.error.errorMessage, 'dismiss');
+      if (error.error) {
+        this.showError(error.error);
       } else if (error.name && error.message) {
         this._snackBar.open(error.name + ': ' + error.message, 'dismiss');
       } else {
-        this._snackBar.open(error.error, 'dismiss');
+        this._snackBar.open('Unknown HTTP error', 'dismiss');
```

Comment:
It looks like these `snakBar` invocations, can be also handled by `showError` if we will add the following condition
```
private showError(error: any): void {
  if (typeof error === 'string') {
     this._snackBar.open(error, 'dismiss')
     return;
  }
  .....
}
```
What do you think?

Reply:
True, that makes it more elegant